src/utils.ts

```diff
@@ -17,42 +18,115 @@ const GA_BASE_URL = `https://www.google-analytics.com/mp/collect?measurement_id=
+        return randomUUID();
+    }
+}
 
-    }).catch(() => {
-        // Fallback to a semi-random ID if machine-id isn't available
-        uniqueUserId = createHash('sha256')
-            .update(`${platform()}-${process.env.USER || process.env.USERNAME || 'user'}-${Date.now()}`)
-            .digest('hex');
-    });
-} catch {
-    // Fallback to a semi-random ID if import fails
-    uniqueUserId = createHash('sha256')
-        .update(`${platform()}-${process.env.USER || process.env.USERNAME || 'user'}-${Date.now()}`)
-        .digest('hex');
+/**
+ * Sanitizes error objects to remove potentially sensitive information like file paths
+ * @param error Error object or string to sanitize
+ * @returns An object with sanitized message and optional error code
+ */
+export function sanitizeError(error: any): {message: string, code?: string} {
+    let errorMessage = '';
+    let errorCode = undefined;
+    
+    if (error instanceof Error) {
+        // Extract just the error name and message without stack trace
+        errorMessage = error.name + ': ' + error.message;
+        
+        // Extract error code if available (common in Node.js errors)
+        if ('code' in error) {
+            errorCode = (error as any).code;
+        }
+    } else if (typeof error === 'string') {
+        errorMessage = error;
+    } else {
+        errorMessage = 'Unknown error';
+    }
+    
+    // Remove any file paths using regex
+    // This pattern matches common path formats including Windows and Unix-style paths
+    errorMessage = errorMessage.replace(/(?:\/|\\)[\w\d_.-\/\\]+/g, '[PATH]');
+    errorMessage = errorMessage.replace(/[A-Za-z]:\\[\w\d_.-\/\\]+/g, '[PATH]');
+    
+    return { 
+        message: errorMessage, 
+        code: errorCode 
+    };
 }
 
 /**
  * Send an event to Google Analytics
  * @param event Event name
  * @param properties Optional event properties
  */
-export const capture = (event: string, properties?: any) => {
-    if (!isTrackingEnabled || !GA_MEASUREMENT_ID || !GA_API_SECRET) {
-        return;
-    }
-    
+export const capture = async (event: string, properties?: any) => {
     try {
+        // Check if telemetry is enabled in config (defaults to true if not set)
+        const telemetryEnabled = await configManager.getValue('telemetryEnabled');
+        
+        // If telemetry is explicitly disabled or GA credentials are missing, don't send
+        if (telemetryEnabled === false || !GA_MEASUREMENT_ID || !GA_API_SECRET) {
+            return;
+        }
+        
+        // Get or create the client ID if not already initialized
+        if (uniqueUserId === 'unknown') {
+            uniqueUserId = await getOrCreateUUID();
+        }
+        
+        // Create a deep copy of properties to avoid modifying the original objects
+        // This ensures we don't alter error objects that are also returned to the AI
+        const sanitizedProperties = properties ? JSON.parse(JSON.stringify(properties)) : {};
```

Comment:
JSON parse better to wrap with try catch 
```suggestion
let sanitizedProperties;
try {
       sanitizedProperties = properties ? JSON.parse(JSON.stringify(properties)) : {};
        } catch () {
        sanitizedProperties = {}
        }
```